packages/react/src/components/Select/Select.tsx

```diff
@@ -107,6 +106,9 @@ const Select = (props: SelectProps) => {
     (option) => option.value === activeOption,
   );
 
+  const resetKeyword = (newValue?: string) =>
+    setKeyword((!multiple && newValue) || '');
+
```

Comment:
This should be wrapped in a [useCallback](https://react.dev/reference/react/useCallback) since you are calling it from a hook. Its also a good habit as React knows it doesn't have to re-declare it every time it re-evaluates the render function.

Reply:
I tend to avoid `useCallback` for simple functions like this. It only makes the code more messy. What is the advantage of using it here? I can't tell if it improves the performance in this case.

Reply:
I agree! It was just to fix the `react-hooks/exhaustive-deps` warning we were getting further down. I've experience several times, bugs that would have been caught if it weren't for a eslint-disabled line 🙈 You never know who is going to come after you and at what level they can understand or code 😅 

Its important to know that function declarations inside a React component always declares a _different_ function on each render, so for the deps array the dependency has changed. Performance wise this function probably isn't measurable, but if you have lots of function declaration in a component and that component i used say 50 or 100 times, you might run into issue. MUI for [example](https://github.com/mui/material-ui/blob/master/packages/mui-base/src/SelectUnstyled/SelectUnstyled.tsx) are quite aggressive with their use of `useMemo/Callback`